nixos/modules/tasks/filesystems.nix

```diff
@@ -354,7 +354,16 @@ in
           options.
         '';
       }
-    ];
+    ] ++ lib.map (fs: {
+      assertion = fs.label != null -> fs.device == "/dev/disk/by-label/${fs.label}";
+      message = ''
+        The filesystem with mount point ${fs.mountPoint} has its label and device set to inconsistent values:
+          label: ${toString fs.label}
+          device: ${toString fs.device}
+        'filesystems.<name>.label' takes priority over device. To resolve this warning, either remove one of
+        the options, or give 'filesystems.<name>.device' a higher priority, e.g. with mkForce.
```

Comment:
It's the other way around, device takes priority over the label, since we assign the device based on the label using `mkDefault`. But it doesn't really matter in the end, since the eval will fail due to the assertion anyway.

To fix the assertion failing, you should either unset (i.e. set to null) the label or the device. It's hard to give general instructions, since it depends where the conflicting values are coming from exactly, but just using mkForce will probably not fix it.

Reply:
In that case, I guess I'll only mention those options are mutually exclusive I did in the initial assert exp.